File: code_examples/4-Functions/5-Default_Values/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program prints out your name, using default function parameters
*/

using System;

/*
 * The Program class
 * Contains all methods for performing how local and global variables work
*/
class Program {

    static string FullName(string firstName, string lastName, string middleName = "") {
        // this function calculates the full name
        string fullName;

        if (middleName == "") {
            fullName = firstName + " " + lastName;
        } else {
            fullName = firstName + " " + middleName + " " + lastName;
        }
        return fullName;
    }

    public static void Main (string[] args) {
        // this function gets a users name and prints out their formal name
        string firstName;
        string question;
        string middleName = "";
        string lastName;
        string fullName;

        //input
        Console.Write("Enter your first name: ");
        firstName = Console.ReadLine();
        Console.Write("Do you have a middle name? (y/n): ");
        question = Console.ReadLine();
        if (question.ToUpper() == "Y" || question.ToUpper() == "YES") {
            Console.Write("Enter your middle name: ");
            middleName = Console.ReadLine();
        }
        Console.Write("Enter your last name: ");
        lastName = Console.ReadLine();
        Console.WriteLine();

        // process & output
        if (middleName != "") {
            fullName = FullName(firstName, lastName, middleName);
        } else {
            fullName = FullName(firstName, lastName);
        }
        Console.WriteLine($"Your formal name is {fullName}.");

        Console.WriteLine ("\nDone.");
    }
}

File: code_examples/4-Functions/5-Default_Values/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module prints out your name, using default function parameters
"""


def full_name(first_name: str, last_name: str, middle_name: str | None = None) -> str:
    """The full_name() function calculates the full formal name, returns str."""
    # return the full formal name

    full_name = first_name
    if middle_name is not None:
        full_name = full_name + " " + middle_name[0] + "."
    full_name = full_name + " " + last_name

    return full_name


def main() -> None:
    """The main() function gets a users name and prints out their formal name, returns None."""

    middle_name = None

    # input & process
    first_name = input("Enter your first name: ")
    question = input("Do you have a middle name? (y/n): ")
    if question.upper() == "Y" or question.upper() == "YES":
        middle_name = input("Enter your middle name: ")
    last_name = input("Enter your last name: ")

    if middle_name is not None:
        name = full_name(first_name, last_name, middle_name)
    else:
        name = full_name(first_name, last_name)

    # output
    print(f"\nYour formal name is {name}.")

    print("\nDone.")


if __name__ == "__main__":
    main()
